src/App.jsx

```diff
@@ -14,45 +16,87 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FORMAT_ALPHABETICALLY = 'alphabet';
+const SORT_FORMAT_BY_LENGTH = 'sortByLength';
+const REVERSE = 'reverse';
+
+function getPreparedGoods(groceries, sortFormat, doReverse) {
+  const preparedGoods = [...groceries];
+
+  if (sortFormat) {
+    preparedGoods.sort(
+      (good1, good2) => {
+        switch (sortFormat) {
+          case SORT_FORMAT_ALPHABETICALLY:
+            return good1.localeCompare(good2);
+
+          case SORT_FORMAT_BY_LENGTH:
+            return good1.length - good2.length;
+
+          default:
+            return 0;
+        }
+      },
+    );
+  }
+
+  return doReverse ? preparedGoods.reverse() : preparedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortFormat, setSortFormat] = useState('');
+  const [doReverse, setDoReverse] = useState('');
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortFormat, doReverse);
+  const HAS_ACTIVE_SORTING = sortFormat !== '' || doReverse !== '';
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${cn({ 'is-light': sortFormat !== SORT_FORMAT_ALPHABETICALLY })}`}
+          onClick={() => setSortFormat(SORT_FORMAT_ALPHABETICALLY)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={`button is-success ${cn({ 'is-light': sortFormat !== SORT_FORMAT_BY_LENGTH })}`}
+          onClick={() => setSortFormat(SORT_FORMAT_BY_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={`button is-warning ${cn({ 'is-light': doReverse !== REVERSE })}`}
+          onClick={() => (doReverse !== ''
+            ? setDoReverse('')
+            : setDoReverse(REVERSE))}
+        >
```

Comment:
You can write handler for it 
And 
```suggestion
          onClick={() => (!doReverse
            ? setDoReverse('')
            : setDoReverse(REVERSE))}
        >
```